target/linux/qualcommax/ipq60xx/base-files/lib/upgrade/platform.sh

```diff
@@ -33,6 +33,20 @@ EOF
 
 platform_do_upgrade() {
 	case "$(board_name)" in
+	linksys,mr7350)
+		boot_part="$(fw_printenv -n boot_part)"
```

Comment:
What about parsing cmdline like that
```
		if [[ $(cat /proc/cmdline) =~ 'ubi.mtd=rootfs' ]]; then
			fw_setenv boot_part 2
			CI_KERNPART="alt_kernel"
			CI_UBIPART="alt_rootfs"
		else
			fw_setenv boot_part 1
			CI_UBIPART="rootfs"
		fi
		fw_setenv boot_part_ready 3
		fw_setenv auto_recovery yes
		nand_do_upgrade "$1"
```

Reply:
I think that sometimes we may intentionally want to change this variable in order to update the current partition.
I'm not sure if a change is needed here.

Reply:
@testuser7 Yeah, agree. And I just copy-pasted the code from ipq807x
https://github.com/openwrt/openwrt/blob/e4453d1f8172feee97e0f66a102aa4e63c47e629/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh#L81

Reply:
@testuser7, I think you  misunderstand my comment. I suggested to not read `boot_part`. It obviously need to be written to ping-pong the boot partition.

@Vladdrako, Thank you for pointing that out. I think that copypasting code is a big no-no because it duplicates any mistakes and issues that already exist, and it creates redundant code.

The preferred procedure would be to share the common functionality. One way is to move this logic to `qualcommax/base-files/` in something named `linksys.sh` and source this file as needed.

if you integrate all my suggestions, you get to something that is symmetrical and doesn't wear out the NAND:

		if <your heuristic here>; then
			new_boot_part=2
			CI_KERNPART="alt_kernel"
			CI_UBIPART="alt_rootfs"
		else
			new_boot_part=1
			CI_KERNPART="kernel"
			CI_UBIPART="rootfs"
		fi
		
		fw_setenv -s - <<-EOF
			boot_part $new_boot_part
			boot_part_ready 3
			auto_recovery yes
		EOF

		nand_do_upgrade "$1"

Reply:
> @testuser7, I think you misunderstand my comment. I suggested to not read `boot_part`. It obviously need to be written to ping-pong the boot partition.

I don't think there's anything wrong with reading the u-boot variable. If someone wants to, they can overwrite them so much that the system won't even start.

Reply:
> I don't think there's anything wrong with reading the u-boot variable. If someone wants to, they can overwrite them so much that the system won't even start.

This has nothing to do with my original comment about using `boot_part` as the heuristic for the installation destination.